training/xgboost/structured/base/README.md

```diff
@@ -0,0 +1,150 @@
+  We recommend to run the scripts in this directory in the following order, and use
+  the `source` command to run them, in order to export the environment variables at each step:
+  * [train-local.sh](./scripts/train-local.sh) trains the model locally using `gcloud`. It is always a
+  good idea to try and train the model locally for debugging, before submitting it to AI Platform.
+  * [train-cloud.sh](./scripts/train-cloud.sh) submits a training job to AI Platform.
+* [setup.py](./setup.py): containing all the required Python packages for this tutorial.
+
+We recommend that you follow the same structure for your own work. In most cases, you only need to 
+modify:
+
+ - `metadata.py`
+ - `model.py`
+ 
+ and leave the other python files untouched.
+
+## Running the Sample
+
+After you go over the steps in the prerequisites section, you are ready to run this sample.
+Here are the steps you need to take:
+
+1. _[Optional]_ Train the model locally. Run:
+ 
+```bash
+source ./scripts/train-local.sh
+``` 
+
+as many times as you like (This has no effect on your cloud usage). If successful, this script should
+create a new model as `trained/quickstart/model.bst`, which means you may now submit a
+training job to AI Platform.
+
+2. Submit a training job to AI Platform. Run: 
+
+```bash
+source ./scripts/train-cloud.sh
+``` 
+This will create a training job on AI Platform and displays some instructions on how to track the job progress.
+At the end of a successful training job, it will uplaod the trained model object to a GCS
+bucket and sets `$MODEL_DIR` environment variable to the directory containing the model.
+
+## Explaining Key Elements
+
+In this section, we'll highlight the main elements of this sample.
+
+### [model.py](trainer/model.py)
+
+In this sample, we simply create an instance of `XGBClassifier` estimator and return it.
+
+### [metadata.py](trainer/metadata.py)
+
+We define which features should be used for training. We also define what the target is.
+
+### [train-local.sh](./scripts/train-local.sh)
+
+The command to run the training job locally is this:
+
+```bash
+gcloud ai-platform local train \
+        --module-name=trainer.task \
+        --package-path=${PACKAGE_PATH} \
+        --job-dir=${MODEL_DIR} \
+        -- \
+        --log-level DEBUG \
+        --input=${TAXI_TRAIN_SMALL} \
+        --n-estimators=20 \
+        --max-depth=3
+```
+
+* `--job-dir`: path for the output artifacts, e.g. the model object
+* `module-name` is the name of the Python file inside the package which runs the training job
+* `package-path` determines where the training Python package is.
+* `--` this is just a separator. Anyhing after this will be passed to the training job as input argument.
+* `log-level` sets the python logger level to DEBUG for this script (default is INFO)
+* `--input`: path to the input dataset
+* `--n-estimators` and `--max-depth`: will be passed to the estimator as hyperparameters
```

Comment:
hyperparameters? On a base sample?

Reply:
Yes. It is not hyperparameter tuning.   :)

Reply:
Ah, ok. Should probably remove that comment then.

Reply:
I mean they are hyperparameters, but we are not tuning them in this sample. These hyperparameters are set to constant values.